File: FrontSPA/src/app/core/models/product.ts
import { ProductColor } from "./ProductColor";
import { ProductImage } from "./productImage";

export interface Product {
    id: string;
    name: string;
    nameAr: string;
    price:number;
    disctription?: string;
    disctriptionAr?: string;
    discount?:number;
    imageMain?:string;
    brand?: string;
    returnDays?:number;
    shippingPrice?:number;
    expireDate?:Date;
    categoryId?: string;
    categoryName?: string;
    categoryNameAr?: string;
    productImages?:ProductImage[];
    ProductColors?:ProductColor[];
    isShop:boolean;


}

File: BackEnd.API/Core/Entities/Product.cs
using System;
using System.Collections.Generic;
using Core.Common;

namespace Core.Entities
{
    public class Product:BaseEntity
    {  
        public string Name { get; set; }
        public string NameAr { get; set; }
        public int Price { get; set; }
        public string Disctription { get; set; }
        public string DisctriptionAr { get; set; }
        public string Brand { get; set; }
        public int ReturnDays { get; set; }
        public int ShippingPrice { get; set; }
        public DateTime ExpireDate { get; set; }
        public Guid CategoryId { get; set; }
        public Category Category { get; set; }
        public ICollection<ProductImage> ProductImages { get; set; }
        public ICollection<ProductColor> ProductColors { get; set; }
        public ICollection<ProductBestSeller> ProductsBestSeller { get; set; }
        public ICollection<ShoppingCart> ShoppingCarts { get; set; }
        public ICollection<ProductOrder> ProductOrders { get; set; }

    }
}